src/chain.ts

```diff
@@ -14,47 +17,60 @@ const createHttpResponse = (statusCode, message) => {
     ].join('\r\n');
 };
 
-/**
- * @typedef Options
- *
- * @property {ClientRequest} request
- * @property {net.Socket} sourceSocket - a stream where to pipe from
- * @property {Buffer} head - optional, the response buffer attached to CONNECT request
- * @property {*} handlerOpts - handler options that contain upstreamProxyUrlParsed
- * @property {http.Server} server - the server that we will use for logging
- * @property {boolean} isPlain - whether to send HTTP CONNECT response
- */
-
-/**
- * @param {Options} options
- */
-const chain = ({ request, sourceSocket, head, handlerOpts, server, isPlain }) => {
+interface Options {
+    method: string;
+    headers: string[];
+    path?: string;
+}
+
+export interface HandlerOpts {
+    upstreamProxyUrlParsed: URL;
+}
+
+export const chain = (
+    {
+        request,
+        sourceSocket,
+        head,
+        handlerOpts,
+        server,
+        isPlain,
+    }: {
+        request: { url?: string },
+        sourceSocket: net.Socket,
+        head?: Buffer,
+        handlerOpts: HandlerOpts,
+        server: EventEmitter & { log: (...args: any[]) => void; },
```

Comment:
`unknown` might be better than `any` here

Reply:
It has to be any, otherwise `server.ts` errors that `unknown` cannot be cast to `string` etc.